File: packages/java/src/test/java/ai/alumnium/system/ObscuredElementTest.java
package ai.alumnium.system;

import static org.assertj.core.api.Assertions.assertThat;

import java.io.File;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;

@DisabledIfEnvironmentVariable(named = "ALUMNIUM_DRIVER", matches = "appium.*")
public class ObscuredElementTest extends BaseTest {

  private static final String OBSCURED_ELEMENT_URL =
      new File("../python/examples/support/pages/obscured_element.html").toURI().toString();

  @Test
  void testClickElementCoveredByStickyBar() {
    navigate(OBSCURED_ELEMENT_URL);
    al.act("click the 'Click Me' button");
    assertThat(al.get("status message")).asString().contains("button clicked");
  }
}

File: packages/python/examples/pytest/obscured_element_test.py
from os import getenv

from pytest import mark


@mark.xfail(
    "appium" in getenv("ALUMNIUM_DRIVER", "selenium"),
    reason="Not supported on Appium driver yet",
)
def test_click_element_covered_by_sticky_bar(al, navigate):
    navigate("obscured_element.html")
    al.do("click the 'Click Me' button")
    assert "button clicked" in al.get("status message")
